Sources/MapboxMaps/Location/LocationProducer.swift

```diff
@@ -142,6 +150,60 @@ internal final class LocationProducer: LocationProducerProtocol {
         syncIsUpdating()
     }
 
+    // MARK: - Location
+
+    private func startUpdatingInterfaceOrientation() {
+        // backup timer if there are some cases when `UIDevice.orientationDidChangeNotification` is not fired
+        // on user interface orientation change
+        // not sure if this is needed at all
+        let backupTimerInterval: TimeInterval = 3
+        headingOrientationUpdateBackupTimer = Timer.scheduledTimer(
+            withTimeInterval: backupTimerInterval,
+            repeats: true
+        ) { [weak self] _ in
+            self?.updateHeadingOrientationIfNeeded(showWarning: true)
+        }
+        headingOrientationUpdateBackupTimer?.tolerance = 0.5
+
+        device.beginGeneratingDeviceOrientationNotifications()
+        notificationCenter.addObserver(self,
+                                       selector: #selector(deviceOrientationDidChange),
+                                       name: UIDevice.orientationDidChangeNotification,
+                                       object: nil)
+    }
+
+    private func stopUpdatingInterfaceOrientation() {
+        headingOrientationUpdateBackupTimer?.invalidate()
+        headingOrientationUpdateBackupTimer = nil
+
+        device.endGeneratingDeviceOrientationNotifications()
+        notificationCenter.removeObserver(self, name: UIDevice.orientationDidChangeNotification, object: nil)
+    }
+
+    @objc private func deviceOrientationDidChange() {
+        updateHeadingOrientationIfNeeded()
+    }
+
+    internal func updateHeadingOrientationIfNeeded() {
+        updateHeadingOrientationIfNeeded(showWarning: false)
+    }
+
+    private func updateHeadingOrientationIfNeeded(showWarning: Bool) {
+        guard let view = userInterfaceOrientationView,
+              let headingOrientation = interfaceOrientationProvider.headingOrientation(for: view) else {
+            return
+        }
+
+        // Setting this property causes a heading update,
+        // so we only set it when it changes to avoid unnecessary work.
+        if locationProvider.headingOrientation != headingOrientation {
+            locationProvider.headingOrientation = headingOrientation
+            if showWarning {
+                Log.warning(forMessage: "Unexpected user interface orientation change was detected. Please file an issue at https://github.com/mapbox/mapbox-maps-ios/issues")
```

Comment:
if I understand correctly, the default value of `locationProvider.headingOrientation` is `portrait`, so when I start an app in other orientation mode, I will receive this warning, is this expected?

Reply:
Good point, I addressed it by updating heading orientation when location producer is getting activated.